### DISCUSSION TITLE
Airbot Omnibus F4 Nano V6 FPORT Telemetry & GPS problem

### DISCUSSION THREAD

**Discussion by cubical23** [iNavFlight/inav#6607] [Q&A] [answered]
Hi everyone,

I have Airbot Omnibus F4 Nano V6 flashed with INAV target FIREWORKSV2. I've connected a FrSky RXSR via FPORT using a bidirectional inverter to UART1 (using RX & TX lines from converter). Softserial is disabled. This setup is working fine on two other F4 flight controllers (Matek F405, Matek411).

I also connected a magnetometer to SPI/UART3 and a UBLOX7 GPS to UART6. 

FPORT & telemetry are working fine with this setup, as long as I don't configure GPS. When GPS is turned on in configurator/cli, I won't get the INAV telemetry data anymore. It does not matter, if GPS wires are pysically connected or not. I also selected a different UART for GPS (without connecting a GPS device), but the problem stays the same: no telemetry when GPS is turned on. 

When (re)booting with GPS turned on, I get telemetry data (even the GPS-related telemetry items) for about 2 seconds, then telemetry is gone again.

So I'm not sure what's wrong. Is this a config problem, a hardware limitation or even a bug?

I attached my current diff for reference.

Thanks in advance and happy flying

[INAV_cli_RetroCopter_20210212_213354.txt](https://github.com/iNavFlight/inav/files/5974348/INAV_cli_RetroCopter_20210212_213354.txt)

**Comment by cubical23** [comment]
I triple checked the wires & connections, so I'm very sure, that everything is fine. Also I could not find any configuration issues. Telemetry is just not working with GPS enabled. GPS works fine.

All UARTS are in use (UART1 for FPORT, UART2 for Tramp, UART3/SPI for Magnetometer, UART4 for Telemetry, UART6 for GPS).

I think, it's a bug, I can't find any other explanation.

**Reply by MRC3742** [reply]
@cubical23 I had the same problem setting up a similar board. RXSR-FC (OMNIBUS F4 Fireworks V2)
I had to change and add sections to the target code as this board and RX combo is currently not supported in iNav.
After getting the board running, telemetry stopped working after turning on the GPS ?
After spending hours researching the problem I found some issues on BF and FrSky github pages.
RXSR firmware update 2.1.1 seems to be causing telemetry loss.
https://github.com/FrSkyRC/Firmware-Test/issues/118

Reverting the RXSR firmware back to version 2.1.0 fixed my no telemetry issue!!!

Hope this helps - Rich

**Reply by cubical23** [reply]
nice finding, I switched back to the older 1.x firmware (2.1.0 seems to be buggy too, according to 2.1.1 redme.txt) and it's working, thanks a lot! I still can't believe that this is the reason.

**Comment by MRC3742** [comment, marked as answer]
Glad to hear that changing RX firmware back to an earlier version got it working for you!
I'm running v2.1.0 with failsafe set to "no pulses" and all seems to be working properly?

Maybe you can mark this discussion as answered to help others with a similar issue.

**Reply by cubical23** [reply]
this is what's in the readme for 2.1.1 fport lbt version:

> v2.1.1
> Fix the issue that no output under failsafe mode （F.port firmware only）

I also tested this, but with "hold" instead of "no pulses", which also worked, at least when turning off the TX (but it felt like it took some time, at least > 1sec). So not sure, what this release message really means. Maybe it even does not matter for inav. AFAIK is there a extra bit in FPORT protocol indicating that there's a failsafe situation.

For me it seems that every version (fport lbt) of the rxsr firmware has an issue. I also had other issues with 2.1.1: two of my planes reported "telemetry lost" all the time. Going back to 1.x also solved this problem.

Also, with latest 1.x  (RXSR-FPORT_LBT_ACCST_191128) I can't bind with my T16 Pro, binding with the previous version (R-XSR_FPORT_LBT_190311) and updating afterwards is working.

btw the release notes of RXSR-FPORT_LBT_ACCST_191128 contains this:
> Fixed the problem of abnormal high frame loss rate.

~I already searched how to mark your answer as the solution, but couldn't find the button :/~

I only could mark your second reply as answer, a comment in response to a comment cannot be marked as answer. So everybody else, see MRC3742's first response and thanks again!
